core/embed/rust/src/trezorhal/bitblt.rs

```diff
@@ -0,0 +1,445 @@
+            BitmapFormat::MONO1P => bitmap.width() as u16, // packed bits
+            _ => bitmap.stride() as u16,
+        };
+
+        Self {
+            // SAFETY:
+            // It's safe if `y` is inside the bitmap (see note above)
+            src_row: unsafe { bitmap.row_ptr(y as u16) },
+            src_stride: bitmap_stride,
+            src_x: x as u16,
+            src_y: y as u16,
+            ..self
+        }
+    }
+
+    /// Sets foreground color used for rectangle filling or
+    /// drawing monochrome bitmaps.
+    fn with_fg(self, fg_color: Color) -> Self {
+        Self {
+            src_fg: fg_color.into(),
+            ..self
+        }
+    }
+
+    /// Sets foreground color used for drawing monochrome bitmaps.
+    fn with_bg(self, bg_color: Color) -> Self {
+        Self {
+            src_bg: bg_color.into(),
+            ..self
+        }
+    }
+
+    /// Sets the foreground alpha value used for rectangle filling or
+    /// bitmap blending.
+    fn with_alpha(self, alpha: u8) -> Self {
+        Self {
+            src_alpha: alpha,
+            ..self
+        }
+    }
+}
+
+/// Rectangle filling operation.
+pub struct BitBltFill {
+    bitblt: ffi::gfx_bitblt_t,
+}
+
+impl BitBltFill {
+    /// Prepares bitblt **fill** operation
+    ///
+    /// - `r` is the rectangle to fill.
+    /// - `clip` is the clipping rectangle and must be completely inside the
+    ///   destination bitmap.
+    /// - `color` is the color to fill the rectangle with.
+    /// - `alpha` is the alpha value to use for blending.
+    ///
+    /// The function ensures proper clipping and returns `None` if the fill
+    /// operation is not needed.
+    pub fn new(r: Rect, clip: Rect, color: Color, alpha: u8) -> Option<Self> {
+        let r = r.clamp(clip);
+        if !r.is_empty() {
+            Some(Self {
+                // SAFETY:
+                // The only unsafe operation is `.with_rect()`, which is safe
+                // as long as the rectangle is completely inside the destination bitmap.
+                // We will set the destination bitmap later, so for now, we can
+                // safely assume that the rectangle is within the bitmap.
+                bitblt: unsafe {
+                    ffi::gfx_bitblt_t::default()
+                        .with_rect(r)
+                        .with_fg(color)
+                        .with_alpha(alpha)
+                },
+            })
+        } else {
+            None
+        }
+    }
+
+    /// Fills a rectangle in the destination bitmap with the specified color.
+    ///
+    /// Destination bitmap must be in RGB565 format
+    pub fn rgb565_fill(&self, dst: &mut Bitmap) {
+        assert!(dst.format() == BitmapFormat::RGB565);
+        // SAFETY:
+        // - The destination bitmap is in the correct format.
+        // - The destination rectangle is completely inside the destination bitmap.
+        unsafe { ffi::gfx_rgb565_fill(&self.bitblt.with_dst(dst)) };
```

Comment:
Yes, you are right; the interface could be improved in that way. However, as we discussed in person, I suggest we leave it as is for now, since implementing such changes could lead to more significant modifications. Let's allow time to test the interface as it currently stands. Perhaps we will come up with a better solution in the future.